pmlc/conversion/pxa_to_affine/pxa_to_affine.cc

```diff
@@ -236,9 +237,16 @@ struct AffineVectorReduceOpConversion
                ConversionPatternRewriter &rewriter) const override {
     auto source = rewriter.create<AffineVectorLoadOp>(
         op.getLoc(), op.getVectorType(), op.mem(), op.idxs());
+    // Get an attribute form of the map
+    auto srcMapAttr = AffineMapAttr::get(op.map());
+    // Set the nap attribute
+    source.setAttr(AffineVectorLoadOp::getMapAttrName(), srcMapAttr);
     auto reduce = createVectorReduction(rewriter, op, source.getResult());
-    rewriter.create<AffineVectorStoreOp>(op.getLoc(), ArrayRef<Type>{}, reduce,
-                                         op.mem(), op.idxs());
+    auto dest = rewriter.create<AffineVectorStoreOp>(
+        op.getLoc(), ArrayRef<Type>{}, reduce, op.mem(), op.idxs());
+    // Set the nap attribute
+    auto destMapAttr = AffineMapAttr::get(op.map());
```

Comment:
Both src + dest take the same map (and thus the same attribute), no need to call AffineMapAttr::get twice.

Reply:
Sure! I had it that way, but I have seen LLVM having issues with sharing state objects like this. I choose to stay on the safe side, but I'll change it.

Reply:
Changed.

Reply:
So attributes + types are interned in MLIR (i.e. they are copied into a big buffer in the MLIR context that is never freed until context destruction) and so they are always safe (and cheap) to reuse aggressively, but constructing them (i.e. interning) is mildly expensive because it involves checking if an identical value exists, and if not, copying and compacting the memory representation across.